includes/sanitizers/class-amp-block-sanitizer.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Class AMP_Block_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Block_Sanitizer
+ *
+ * Modifies elements created as blocks to match the blocks' AMP-specific configuration.
+ */
+class AMP_Block_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Tag.
+	 *
+	 * @var string Figure tag to identify wrapper around AMP elements.
+	 * @since 1.0
+	 */
+	public static $tag = 'figure';
+
+	/**
+	 * AMP elements.
+	 *
+	 * @todo Add all elements.
+	 * @var array AMP elements to add layout to.
+	 */
+	public static $amp_embeds = array(
+		'amp-youtube',
+		'amp-facebook',
+		'amp-twitter',
+		'amp-vimeo',
+		'amp-instagram',
+		'amp-dailymotion',
+		'amp-hulu',
+		'amp-reddit',
+		'amp-soundcloud',
+	);
+
+	/**
+	 * Sanitize the AMP elements contained by <figure> element where necessary.
+	 *
+	 * @since 0.2
+	 */
+	public function sanitize() {
+		$nodes     = $this->dom->getElementsByTagName( self::$tag );
+		$num_nodes = $nodes->length;
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+			$node = $nodes->item( $i );
+
+			// We're looking for <figure> elements that have one child node only.
+			if ( 1 !== count( $node->childNodes ) ) {
+				continue;
+			}
+
+			$attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node );
+
+			// We are looking for <figure> elements with layout attribute only.
+			if ( ! isset( $attributes['data-amp-layout'] ) && ! isset( $attributes['data-amp-noloading'] ) ) {
+				continue;
+			}
+
+			$amp_el_found = false;
+
+			foreach ( $node->childNodes as $child_node ) {
+				if ( ! in_array( $child_node->tagName, self::$amp_embeds, true ) ) {
+					continue;
+				}
+				$amp_el_found = true;
+
+				if ( isset( $attributes['data-amp-layout'] ) ) {
+					$child_node->setAttribute( 'layout', $attributes['data-amp-layout'] );
+				}
+				if ( isset( $attributes['data-amp-noloading'] ) && true === filter_var( $attributes['data-amp-noloading'], FILTER_VALIDATE_BOOLEAN ) ) {
+					$child_node->setAttribute( 'noloading', '' );
```

Comment:
Instead of setting the attributes here, why not set them in the `set_layout_attributes` method?